### DISCUSSION TITLE
Try and Catch Blocks

### DISCUSSION THREAD

**Discussion by MoonlitBeans** [jgoll7/Tafe22S211#2] [General]
Have you been using try and catch blocks to ensure that only text/numbers can be entered into specific text boxes?

I can provide example code if you need it. :)

**Comment by jgoll7** [comment]
Memory is a bit foggy but that is for input validation right? If so, I have not used added any so far but can definitely do that.

Example code would be appreciated just to make sure that I'm thinking of the same thing!

**Comment by MoonlitBeans** [comment]
Input Validation is the right name! I couldn't remember. I don't think its required, but it does make sense to have it in. But its one of the only things I can think of for six discussion topics. :l

try
{
		inputValue = double.Parse(valueInputTextBox.Text.ToString());
}

catch
{
		var dialogMessage = new MessageDialog("You cannot convert non numeric values.");
		await dialogMessage.ShowAsync();
		valueInputTextBox.Focus(FocusState.Programmatic);
		return;
}

**Reply by jgoll7** [reply]
Yeah definitely makes sense, I'll be sure to add it into my code!

I haven't done too much on mine yet, hoping to get it going tonight. How is your feature coming along?

**Reply by jgoll7** [reply]
oh do we need to start 3 different discussion threads each? or can we just ask questions in here?

**Reply by MoonlitBeans** [reply]
It's done. I had a little bit of trouble since the text box is classed a string and all the calculations are done in double/int.

**Reply by MoonlitBeans** [reply]
> oh do we need to start 3 different discussion threads each? or can we just ask questions in here?

New threads I think. We both need to make three and reply to three.

**Reply by jgoll7** [reply]
ahhh right that could be tricky
